### DISCUSSION TITLE
Why are formatting options so limited

### DISCUSSION THREAD

**Discussion by jaschaio** [rome/tools#3698] [General]
Hey there,
I am currently using eslint but it's slow. Using it in combination with esbuild my once < 1s build's now went up to 3 - 4s again which is just frustrating.

Rome sounds like a good alternative, but I find the formatting options too limited to be practical.

The documentation says about Formatting:

> only supports few options to avoid that debates over styles turn into debates over Rome options.

Where can I find more information or discussion that lead to that decission?

I am coming from the WordPress world which has pretty opinionated formatting rules as well, for example [for extra spacing within functions, arrays and objects](https://developer.wordpress.org/coding-standards/wordpress-coding-standards/javascript/#arrays-and-function-calls):

```js
array = [ a, b ];

foo( arg );

foo( 'string', object );

foo( options, object[ property ] );

foo( node, 'property', 2 );

prop = object[ 'default' ];

firstArrayElement = arr[ 0 ];
```

While I understand that this is uncommon, because of WordPress sheer scale it is still a common practice in a very big amount of codebases. Are these simply excluded from ever using rome?

Thanks

**Comment by MichaReiser** [comment]
Rome follows a similar [philosophy to Prettier's](https://prettier.io/docs/en/option-philosophy.html). 

The goal is that people spend less time talking about how code should be formatted and we believe that adding many options simply shifts the conversations from "how to format your code" to "what formatting options to".

**Reply by jaschaio** [reply]
Thanks for the context. I love the [linked issue](https://github.com/prettier/prettier/issues/40) and  as well the [linked Wikipedia article](https://en.m.wikipedia.org/wiki/Law_of_triviality). Both are things that immediately resonante. It might be worth to add these to the Rome documentation for context on this decision and philosophy.

But does this basically mean: once you adopt Rome you simply need to abandon whatever formatting rules you are used to and get used to whatever the standard of rome or prettier is. Or is this the main reason why people might not adopt Rome/Prettier because it's to big of a mind change to make?

**Reply by jeysal** [reply]
+1 on the docs change, would you like to contribute a PR with it?

> you simply need to abandon whatever formatting rules you are used to and get used to whatever the standard of rome or prettier is.

I think it is fortunate that Prettier is already so popular (and still growing in use it seems), so a big chunk of JS developers will already be used to the same formatting rules as Rome's. Some of the most-requested options will be implemented, like for [semicolons](https://github.com/rome/tools/pull/3702). But yeah I think for people who are really passionate about a specific way of formatting, catering to them all increases complexity a lot, so it's probably better for these to stick with the tools that they already happily use :)

> once you adopt Rome

Perhaps to clarify: Overall adoption of Rome is possible while still using a different formatter of preference. While using everything Rome provides out of the box is supposed to be a smooth and fast experience, I don't think Rome will ever force you to use its formatting in order to use other Rome things like the linter.

**Reply by jaschaio** [reply]
@jeysal done: https://github.com/rome/tools/pull/3714
